### DISCUSSION TITLE
Can I configure the retention period for backups in remote storage?

### DISCUSSION THREAD

**Discussion by Programmeris** [Altinity/clickhouse-backup#612] [Q&A] [answered]
I have a Clickhouse cluster that I back up with clickhouse-backup. Backups are uploaded to remote S3(Minio).I need to store backups for a limited time, for example, 1 month. Does clickhouse-backup have built-in backup rotation?

**Comment by Slach** [comment, marked as answer]
yes, currently only simple rotation "keep last X backup" implements

look https://github.com/AlexAkulov/clickhouse-backup/#default-config

if you backup every day
you can use /etc/clickhouse-backup/config.yml
```yaml
general:
  keep_backup_remote: 30
  keep_backup_local: 1 
```

moreover, you need to call `clickhouse-backup delete local backup_name` after successful `clickhouse-backup create_remote backup_name` to avoid allocate additional disk space after original data parts will delete during background data part merges

**Comment by Programmeris** [comment]
@Slach Thanks for the answer! But I have two more questions.

First. Are the `system.backup_actions` and `system.backup_list` tables cleared automatically when clickhouse-backup is deployed in API server  mode (like Kubernetes)?

Second. How to call `clickhouse-backup delete local backup_name `in Kubernetes? Something like `clickhouse-client --echo -mn -q "INSERT INTO system.backup_actions(command) VALUES('delete ${SERVER}-${BACKUP_NAMES[$SERVER]}')" --host="$SERVER " --port="$CLICKHOUSE_PORT" --user="$BACKUP_USER" $BACKUP_PASSWORD`? I am using manifests from here https://github.com/AlexAkulov/clickhouse-backup/blob/master/Examples.md#how-to-use-clickhouse-backup-in-kubernetes.

**Comment by Slach** [comment]
> First. Are the system.backup_actions and system.backup_list tables cleared automatically when clickhouse-backup is deployed in API server mode (like Kubernetes)?

`system.backup_actions` - ENGINE=URL() which represents clickhouse-backup internal actions list memory structure. This structure is not persistent, after restart clickhouse-backup is creates from scratch.

`system.backup_list` - also ENGINE=URL() and represents current local and remote backups list

> Second. How to call clickhouse-backup delete local backup_name in Kubernetes? Something like clickhouse-client --echo -mn -q "INSERT INTO system.backup_actions(command) VALUES('delete ${SERVER}-${BACKUP_NAMES[$SERVER]}')" --host="$SERVER " --port="$CLICKHOUSE_PORT" --user="$BACKUP_USER" $BACKUP_PASSWORD? 

yes, this is properly command

**Reply by Programmeris** [reply]
Thanks
